src/modules/Comment/controller.ts

```diff
@@ -0,0 +1,58 @@
+import type { Request, Response } from 'express';
+import { AppDataSource } from '../../config/database.js';
+import { News } from '../../entities/News.js';
+import { Comment } from '../../entities/Comment.js';
+
+const commentRepository = AppDataSource.getRepository(Comment);
+const newsRepository = AppDataSource.getRepository(News);
+
+const commentOnNewsArticle = async (req: Request, res: Response) => {
+  const { user } = req;
+  const { title, description, source, url, publishedAt, content, timeStamp } =
+    req.body;
+
+  if (!user) {
+    res.status(401).send({ message: 'Unauthorized' });
+    return;
+  }
+  if (!content || typeof content !== 'string' || content.length === 0) {
+    res.status(400).send({ message: 'Comment content is required' });
+    return;
+  }
+
+  try {
+    let newsArticle = await newsRepository.findOneBy({ url });
+
+    if (!newsArticle) {
+      newsArticle = newsRepository.create({
+        url,
+        publishedAt,
+        source,
+        description,
+        title,
+      });
+      await newsRepository.save(newsArticle);
+    }
+
+    const comment = commentRepository.create({
+      content,
+      timeStamp: timeStamp ? new Date(timeStamp) : null,
+      user,
+      news: newsArticle,
+    });
+
+    await commentRepository.save(comment);
+
+    res.status(201).send({
+      content: comment.content,
```

Comment:
save the full comment, so it is possible to retrieve everything related to it

Reply:
The full comment is saved in line 44. And just the comment.content send in the response.

Reply:
@MariemSo @Francieli-S yes, the full comment is saved, but not fully sent to the client. and just need to be edited to same only the userId and not the whole user.